pkg/cloud/aws/services/iam/types.go

```diff
@@ -82,22 +83,25 @@ type Resources []string
 type PrincipalID []string
 
 // Conditions is the map of all conditions in the statement entry.
-type Conditions map[string]ConditionValues
-
-// ConditionValues are a list of condition values in a condition statement
-type ConditionValues []string
+type Conditions map[string]interface{}
 
 // JSON is the JSON output of the policy document
 func (p *PolicyDocument) JSON() (string, error) {
 	b, err := json.Marshal(&p)
 	if err != nil {
 		return "", err
 	}
-	return string(b), nil
+	return ProcessPolicyDocument(string(b)), nil
 }
 
 // NewManagedName creates an IAM acceptable name prefixed with this Cluster API
 // implementation's prefix.
 func NewManagedName(prefix string) string {
 	return fmt.Sprintf("%s.%s", prefix, IAMSuffix)
 }
+
+// ProcessPolicyDocument replaces IAMConditions with Condition to work around
+// GoFormation
```

Comment:
Can we add a little bit more context on why this is needed?

Reply:
Extended the comment on both that function and the `StatementEntry` type.